### PR TITLE
ctb: Add unit tests from SystemConfig

### PR DESCRIPTION
**Pull request by maurelian**
**Description**

Add System Config tests.

**Metadata**
- ENG-2949

### CODE REVIEW

**Comment by mslipper** [comment]
Looks like we're missing the gas snapshot.

File: packages/contracts-bedrock/contracts/test/SystemConfig.t.sol [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+
+        vm.expectEmit(true, true, true, true);
+        emit ConfigUpdate(
+            0,
+            SystemConfig.UpdateType.GAS_CONFIG,
+            abi.encode(newOverhead, newScalar)
+        );
+
+        vm.prank(alice);
+        sysConf.setGasConfig(newOverhead, newScalar);
```
Comment (clabby) [inline packages/contracts-bedrock/contracts/test/SystemConfig.t.sol:87]: Add check that the `overhead` + `scalar` was set correctly

```suggestion
        sysConf.setGasConfig(newOverhead, newScalar);
        assertEq(sysConf.overhead(), newOverhead);
        assertEq(sysConf.scalar(), newScalar);
```

**Review by clabby — CHANGES_REQUESTED** [review]
Looks good. For completeness, may want to `vm.expectEmit` the `ConfigUpdate` event in the fuzz tests as well, but not necessary.

**Review by clabby — APPROVED** [review]

File: packages/contracts-bedrock/contracts/test/SystemConfig.t.sol [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+contract SystemConfig_Initialize_TestFail is CommonTest {
+    function test_initialize_lowGasLimit_reverts() external {
+        vm.expectRevert("SystemConfig: gas limit too low");
+        new SystemConfig({
+            _owner: alice,
+            _overhead: 0,
+            _scalar: 0,
+            _batcherHash: bytes32(hex""),
+            _gasLimit: 7_999_999
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/test/SystemConfig.t.sol:29]: I don't love how `7_999_999` is a magic value - it should have a comment as to how this value was picked

File: packages/contracts-bedrock/contracts/test/SystemConfig.t.sol [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+    function testFuzz_setGasConfig_succeeds(uint256 newOverhead, uint256 newScalar) external {
+        vm.expectEmit(true, true, true, true);
+        emit ConfigUpdate(
+            0,
+            SystemConfig.UpdateType.GAS_CONFIG,
+            abi.encode(newOverhead, newScalar)
+        );
+
+        vm.prank(alice);
```
Comment (tynes) [inline packages/contracts-bedrock/contracts/test/SystemConfig.t.sol:75]: nit: `vm.prank(sysConf.owner())`

Makes it less contextual when reading and decouples the deploy constructor args

**Review by tynes — APPROVED** [review]

**Review by smartcontracts — APPROVED** [review]

File: packages/contracts-bedrock/contracts/test/SystemConfig.t.sol (hunk 3 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/SystemConfig.t.sol]: good call

File: packages/contracts-bedrock/contracts/test/SystemConfig.t.sol (hunk 2 above)
Comment (maurelian) [reply packages/contracts-bedrock/contracts/test/SystemConfig.t.sol]: ok fixed.

**Pull request merged**